sos/report/plugins/openstack_instack.py

```diff
@@ -81,16 +81,17 @@ def setup(self):
             # capture all the possible stack uuids
             get_stacks = "openstack stack list"
             stacks = self.collect_cmd_output(get_stacks)['output']
-            stack_ids = re.findall(r'(\s(\w+-\w+)+\s)', stacks)
+            stack_ids = re.search(r'\|\s(((\w+-){4})\w+)\s\|', stacks)
             # get status of overcloud stack and resources
             for sid in stack_ids:
+                sid = sid[1]
                 self.add_cmd_output([
-                    "openstack stack show %s" % sid[0],
-                    "openstack stack resource list -n 10 %s" % sid[0]
+                    "openstack stack show %s" % sid,
+                    "openstack stack resource list -n 10 %s" % sid
                 ])
 
                 # get details on failed deployments
-                cmd = "openstack stack resource list -f value -n 5 %s" % sid[0]
+                cmd = "openstack stack resource list -f value -n 5 %s" % sid
```

Comment:
Why the change from `sid[0]` to `sid[1]` value in commands arguments?

Reply:
The new regex groups the matches differently - I grabbed some output from some docs and built against, but it would be great if we could have some "real" validation on it.

Reply:
Ah I see - but why `re.search` (that searches for the first instance only) when we need to spot all stack IDs?

Moreover, the code doesnt work for me on an an example output like:

```
+--------------------------------------+------------+-----------------+---------------------+--------------+
| ID                                   | Stack Name | Stack Status    | Creation Time       | Updated Time |
+--------------------------------------+------------+-----------------+---------------------+--------------+
| 70b9feca-8f99-418e-b2f1-cc38d61b3ffb | MYSTACK    | CREATE_COMPLETE | 2016-06-08T09:54:15 | None         |
| 82f75a62-9e03-4031-afb1-a63022f8dd68 | SOMESTACK  | ALSO_COMPLETE   | 2022-11-21T08:46:01 | None         |
+--------------------------------------+------------+-----------------+---------------------+--------------+
```

where the `for sid in stack_ids:` fails with `TypeError: '_sre.SRE_Match' object is not iterable`.

Reply:
Good catch. I was overlooking this was scanning the entire output, not line by line like our other matching methods, so this should go back to `findall()`.